cluster-agent/utils/retrieve_metrics.go

```diff
@@ -0,0 +1,55 @@
+package utils
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+// GET Request to retrive API Server metrics from the route endpoint
+func GetAPIServerMetrics(routeEndpoint string) string {
+	// #nosec G107
+	resp, err := http.Get(routeEndpoint)
+	if err != nil {
```

Comment:
For these 3 Get functions, you can return `(string, error)` from the func, and instead `return "", err` if the err is non-nil (rather than just logging it via `log.Fatalln`).